src/java/org/apache/cassandra/index/sai/plan/TopKProcessor.java

```diff
@@ -244,7 +269,7 @@ private <U extends Unfiltered, R extends BaseRowIterator<U>, P extends BaseParti
                     {
                         PartitionResults pr = processPartition(partitionRowIterator);
                         topK.addAll(pr.rows);
-                        for (var uf: pr.tombstones)
+                        for (var uf : pr.tombstones)
                             addUnfiltered(unfilteredByPartition, pr.partitionInfo, uf);
```

Comment:
It's not obvious to me that `unfilteredByPartition != null` here. There is no null check above and it looks like this whole `else` branch may run if `partitions instanceof PartitionIterator` in which case we don't initialize `unfilteredByPartition`. Is it a bug or am I missing something? If not a bug, can you explain why NPE won't happen here?

Reply:
`unfilteredByPartition` can be null only if we are sorting a `PartitionIterator`. I think in that case the `PartitionResults` shouldn't contain any tombstones, so there won't be any iteration in this loop. I can anyway add an `if` or an assertion if it makes it clearer/safer.

Reply:
Yeah, please add an assertion.

Reply:
+1, just added an assertion.